### ISSUE TITLE
Random generator produces limited amount of random data

### ISSUE THREAD

**Issue by Snawoot** [rfk/pypyjs#93]
Random generator produces limited amount of random data.
How to reproduce:
```python
import random
a = [ random.randint(0, 1000000) for _ in xrange(1050) ]
print a
```
In this example all values after 1042nd element will be equal zero

**Comment by rfk** [comment]
Thanks for the report!  This seems limited to the default `SystemRandom` class, which in theory is built upon `crypto.getRandomValues` in the browser:
```
>>> s = random.SystemRandom()
>>> a = [s.randint(0, 1000000) for _ in xrange(1050) ]
>>> a.index(0)
1042
>>> a[-20:]
[664355, 356034, 321203, 517022, 396342, 122604, 244866, 187161, 936225, 131769, 78809, 628629, 0, 0, 0, 0, 0, 0, 0, 0]
>>> 
```
Versus:
```
>>> w = random.WichmannHill()
>>> a = [ w.randint(0, 1000000) for _ in xrange(1050) ]
>>> a.index(0)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ValueError: 0 is not in list
```

What's more, `SystemRandom` will occasionally produce non-zero values after triggering the described behaviour:
```
>>> a = [s.randint(0, 1000000) for _ in xrange(1050) ]
>>> a.index(0)
66
>>> a.count(0)
982
>>> 
```

Very strange indeed.

**Comment by rfk** [comment]
After adding some `print`s in the source of random.py, it appears that the random number generator is correctly producing random values, but they're winding up in the list as zeroes.  So I wonder if there's a bug in the JIT backend that's failing to populate the list after some amount of warmup in the loop.

**Comment by rfk** [comment]
Yep, trying the same code in a JIT-less PyPy.js behaves correctly, with lots of non-zero integers in the list.  Certainly looks like a JIT code-generation bug.
